src/lib.rs

```diff
@@ -273,35 +280,39 @@ impl Future for HttpConnecting {
 
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
         loop {
-            let state;
-            match self.state {
-                State::Lazy(ref executor, ref mut host, port, local_addr) => {
+            self.state = match mem::replace(&mut self.state, State::Uninitialized) {
+                State::Uninitialized => panic!("Should not be State::Uninitialized!"),
```

Comment:
This is going to get hit when a future is not ready. It would be nice to keep the previous state management as it was a bit cleaner. Maybe we can stuff the host into an Rc?